src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestBindCallGen.generated.txt

```diff
@@ -1,122 +1,139 @@
-
-        System.Collections.Generic.Dictionary<string, string> temp3 = obj.MyDictionary;
-        temp3 ??= new System.Collections.Generic.Dictionary<string, string>();
-        BindCore(configuration.GetSection("MyDictionary"), ref temp3);
-        obj.MyDictionary = temp3;
-
-        System.Collections.Generic.Dictionary<string, Program.MyClass2> temp4 = obj.MyComplexDictionary;
-        temp4 ??= new System.Collections.Generic.Dictionary<string, Program.MyClass2>();
-        BindCore(configuration.GetSection("MyComplexDictionary"), ref temp4);
-        obj.MyComplexDictionary = temp4;
+    public static void Bind(this global::Microsoft.Extensions.Configuration.IConfiguration configuration, global::Program.MyClass obj) => global::Microsoft.Extensions.Configuration.Binder.SourceGeneration.Helpers.BindCore(configuration, ref obj);
+}
 
-    }
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
+{
+    using System;
+    using System.Linq;
+    using Microsoft.Extensions.Configuration;
+    using System.Collections.Generic;
 
-    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref System.Collections.Generic.List<int> obj)
+    internal static class Helpers
     {
-        if (obj is null)
+        public static void BindCore(IConfiguration configuration, ref List<Int32> obj)
         {
-            throw new global::System.ArgumentNullException(nameof(obj));
-        }
+            if (obj is null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
 
-        int element;
-        foreach (Microsoft.Extensions.Configuration.IConfigurationSection section in configuration.GetChildren())
-        {
-            if (section.Value is string stringValue5)
+            Int32 element;
+            foreach (IConfigurationSection section in configuration.GetChildren())
             {
-                element = int.Parse(stringValue5);
-                obj.Add(element);
+                if (section.Value is string stringValue0)
+                {
+                    element = int.Parse(stringValue0);
+                    obj.Add(element);
+                }
             }
         }
-    }
 
-    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref System.Collections.Generic.Dictionary<string, string> obj)
-    {
-        if (obj is null)
+        public static void BindCore(IConfiguration configuration, ref Dictionary<String, String> obj)
         {
-            throw new global::System.ArgumentNullException(nameof(obj));
-        }
+            if (obj is null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
 
-        string key;
-        foreach (Microsoft.Extensions.Configuration.IConfigurationSection section in configuration.GetChildren())
-        {
-            if (section.Key is string stringValue6)
+            String key;
```

Comment:
```suggestion
            string key;
```

See https://blog.paranoidcoding.com/2019/04/08/string-vs-String-is-not-about-style.html?utm%5Fcampaign=featured&utm%5Fmedium=email&utm%5Fsource=csharpdigest

I believe we use `ToDisplayString` in the RequestDelegateGenerator to generate type names.

Reply:
Yeah, this just needs some special casing for primitives; added in next commit.